### ISSUE TITLE
RNA-Seq mode and disambiguation

### ISSUE THREAD

**Issue by schelhorn** [chapmanb/bcbio-nextgen#1171]
It seems as if since STAR transcriptome alignment was re-enabled we lost the ability for disambiguation of these transcriptome alignments that the previous bowtie transcriptome alignment in bcbio.pipeline.rnaseq had. Now, the bowtie disambiguation part is never run if a STAR transcriptome BAM was generated, so Bcbio just takes the non-disambiguated STAR BAM for quantitation, which breaks expectations. It would be really helpful if that could be fixed in the new year - thanks!

**Comment by schelhorn** [comment]
Sorry to bounce this, but is there any way this could be addressed? We had the ability for disambiguation with eXpress (using Bowtie2 realignment) until STAR transcriptome alignment had been introduced, which disabled working with disambiguation reads. I know that Sailfish is the new method for disambiguation, but we haven't properly validated if the disambiguated Sailfish quantitations using the combined index correspond to the much better understood and tested STAR-disambiguated transcriptome alignments using Bowtie2. I think it would be important to do so, but for this we need properly disambiguated transcriptome alignments using STAR (or STAR+bcbio) back in bcbio. Once we have that I'll validate that Sailfish is a sound choice in that regard.

**Comment by roryk** [comment]
I'm so sorry Sven-Eric, this totally fell off my radar. I'll have some time to fix this this weekend.

**Comment by schelhorn** [comment]
Great, thanks a bunch Rory!

**Comment by roryk** [comment]
Getting there, should be all set soon.

**Comment by roryk** [comment]
Hi Sven-Eric,

I fixed it to always use your trick to get the disambiguated transcriptome alignments by regenerating the FASTQ files from the disambiguated BAM file. The STAR transcriptome file causes some issues with the disambiguation script, it is missing some of the required tags from some of the reads. I pinged Alex about it but until he fixes that up we can go with this solution if that works for you.

I have a branch that is saved that does the transcriptome disambiguation at the same time the full alignment disambiguation happens, but will wait on an upstream fix to swap over to that.
